src/dbnode/storage/index/postings_list_cache_lru.go

```diff
@@ -208,14 +200,13 @@ func (c *postingsListLRU) removeElement(e *list.Element) {
 	entry := e.Value.(*entry)
 
 	if patterns, ok := c.items[entry.uuid.Array()]; ok {
-		key := newPatternAndPatternType(entry.pattern, entry.patternType)
-		delete(patterns, key)
+		delete(patterns, newKey(entry.query, entry.patternType))
```

Comment:
nit: for consistency maybe pull it into a variable?  `newKey := newKey(entry.query, entry.patternType)`